aea/crypto/initialiser.py

```diff
@@ -0,0 +1,79 @@
+# -*- coding: utf-8 -*-
+
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""Module wrapping all the public and private keys cryptography."""
+
+from typing import cast, Optional
+
+from aea.crypto.base import Crypto
+from aea.crypto.fetchai_base import FetchCrypto
+from aea.crypto.ethereum_base import EthCrypto
+from aea.crypto.helpers import _try_validate_private_key_pem_path, _create_temporary_private_key_pem_path
+
+
+class Wallet(object):
+    """Store all the public keys we initialise."""
+
+    def __init__(self, private_key_pem_path: Optional[str] = None):
+        """Instantiate a wallet object."""
+        self.crypto_objects = {
+            "default": self._setup_crypto(private_key_pem_path),
+            "fetchai": self._setup_fetch_crypto(),
+            "ethereum": self._setup_ethereum_crypto()
+        }
+
+        self.public_keys = {
+            "default": self.crypto_objects['default'].public_key,
+            "fetchai": self.crypto_objects['fetchai'].public_key,
+            "ethereum": self.crypto_objects['ethereum'].public_key
+        }
+
+        self.private_key_pem_path = ""
+
+    def _setup_crypto(self, private_key_pem_path: Optional[str] = None):
+        """Create the crypto object."""
+        private_key_pem_path = cast(str, private_key_pem_path)
+        if private_key_pem_path == "":
```

Comment:
we could move that logic into crypto to mirror the below functions for ethereum and fetchai

Reply:
I would prefer to have the logic of setting up the cryptos somewhere else rather than wrap everything inside crypto. It will make it easier to add more cryptos.

Reply:
I am not saying to move the `_setup_crypto` method out. I am saying to make it look similar to the other `_setup_x` methods by moving the first lines of the method into crypto (in fact, it might already be there).

Reply:
Yes, sorry I get it now. This code doesn't exist in the crypto. Since it is extracted from the run.py. But I'll move it inside the crypto to match the other two.